src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,239 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+
+    private static RegistrationService service;
+    private static StorageDao dao;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+        dao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userNotNull_Ok() {
+        User user = new User("121212", "323123", 43);
+        dao.add(user);
+        User u = dao.get(user.getLogin());
+        assertEquals(u, user);
+        assertTrue(Storage.people.contains(user));
+    }
+
+    @Test
+    void register_userNull_notOk() {
+        User u = new User(null, null, null);
```

Comment:
never use one symbol or digit for naming
```suggestion
        User user = new User(null, null, null);
```